core/src/main/java/com/backblaze/b2/client/structures/B2UploadProgress.java

```diff
@@ -14,8 +14,35 @@
  *       For example, if we're adding a SHA1 to the thing being uploaded,
  *       the contentLength will be at least 40 bytes longer than you expect
  *       because we're adding an ascii-hex-encoded SHA1.
+ *
+ * NOTE: when storing a large file that involves copying parts, it no
+ *       longer becomes possible to reliably compute the startByte for
+ *       parts that come after a copied part. As a result, these values
+ *       will be set to the constant UNKNOWN_PART_START_BYTE.
+ *
+ *       Additionally, until copied parts are successfully stored and the
+ *       B2 API returns the corresponding B2Part, progress events for copied
+ *       parts will show the length as UNKNOWN_PART_SIZE_PLACEHOLDER. Once a copy succeeds,
+ *       the progress event for that success will contain the correct length.
  */
 public class B2UploadProgress {
+
+    /**
+     * Placeholder value for part sizes that will not be known until the
+     * part is stored in B2. For copy operations, we will not always know
+     * the true part size until we receive the B2Part from b2_copy_part.
+     */
```

Comment:
noe that since no part can be empty, this is always right or an underestimate?  :)

Reply:
Technically a part cannot be less than 5MB, so I strictly said it could be an estimate :)

Reply:
The last part is allowed to be 1 byte.

Reply:
Good point, I forgot about that.